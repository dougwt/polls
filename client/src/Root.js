import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import promiseMiddleware from 'redux-promise';
import reduxThunk from 'redux-thunk';
import reducers from './reducers';

const localAuth = JSON.parse(localStorage.getItem('auth') || null);

const Root = ({
  children,
  initialState = { auth: localAuth ? localAuth : false }
}) => {
  const store = createStore(
    reducers,
    initialState,
    composeWithDevTools(applyMiddleware(reduxThunk, promiseMiddleware))
  );

  return <Provider store={store}>{children}</Provider>;
};
Root.propTypes = {
  children: PropTypes.object.isRequired,
  initialState: PropTypes.object
};

export default Root;
