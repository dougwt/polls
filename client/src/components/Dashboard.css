.tabs {
  background: none;
}
.tabs .tab a,
.tabs .tab a.active,
.tabs .tab a.active:hover {
  color: #06bad6;
}
.tabs .tab a:hover {
  color: #ef476f;
}
.tabs .indicator {
  background-color: #06bad6;
}

.pagination li.active {
  background-color: #06bad6;
}
