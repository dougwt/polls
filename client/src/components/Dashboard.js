import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Section, Container, Row } from 'react-materialize';
import PollList from './polls/PollList';

export class Dashboard extends Component {
  componentDidMount() {
    window.$('ul.tabs').tabs();
  }

  render() {
    return (
      <Section>
        <Container>
          <Row>
            <div className="col s12">
              <h3>Dashboard</h3>
              <ul className="tabs">
                <li className="tab col s3">
                  <a href="#my">My Polls</a>
                </li>
                <li className="tab col s3">
                  <a href="#other">Other Polls</a>
                </li>
              </ul>
            </div>
          </Row>

          <Row>
            <div id="my" className="col s12">
              <PollList
                title="This list contains all the polls created by you."
                polls={this.props.polls}
              />
            </div>
            <div id="other" className="col s12">
              <PollList title="This list contains all the polls created by other users." />
            </div>
          </Row>
        </Container>
      </Section>
    );
  }
}
Dashboard.propTypes = {
  polls: PropTypes.object
};

function mapStateToProps(state) {
  return {
    polls: state.polls
  };
}

export default connect(mapStateToProps)(Dashboard);
