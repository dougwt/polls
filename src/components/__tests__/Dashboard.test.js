import React from 'react';
import { mount, shallow } from 'enzyme';
import { Dashboard } from '../Dashboard';
import Root from '../../Root';

describe('Dashboard', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      polls: {},
      fetchPolls: jest.fn()
    };
    wrapper = shallow(<Dashboard {...props} />);
  });

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains a page title', () => {
    expect(wrapper.find('h3').exists()).toBe(true);
  });

  it('contains a `My Polls` PollList', () => {
    expect(wrapper.find('#my > Connect(PollList)').length).toEqual(1);
  });

  it('contains a `Other Polls` PollList', () => {
    expect(wrapper.find('#other > Connect(PollList)').length).toEqual(1);
  });

  it('calls the fetchPolls action creator', () => {
    global.$ = () => {
      return {
        tabs: jest.fn()
      };
    };
    wrapper = mount(
      <Root>
        <Dashboard {...props} />
      </Root>
    );

    expect(props.fetchPolls).toHaveBeenCalled();
  });
});
