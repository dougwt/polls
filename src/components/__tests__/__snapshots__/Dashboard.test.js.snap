// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    fetchPolls={[MockFunction]}
    polls={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <Row>
          <h3>
            Dashboard
          </h3>
          <ul
            className="tabs"
          >
            <li
              className="tab col s3"
            >
              <a
                href="#my"
              >
                My Polls
              </a>
            </li>
            <li
              className="tab col s3"
            >
              <a
                href="#other"
              >
                Other Polls
              </a>
            </li>
          </ul>
        </Row>
        <Row>
          <div
            className="col s12"
            id="my"
          >
            <Connect(PollList)
              itemsPerPage={15}
              maxButtons={7}
              polls={Array []}
              title="Polls created by you:"
            />
          </div>
          <div
            className="col s12"
            id="other"
          >
            <Connect(PollList)
              itemsPerPage={15}
              maxButtons={7}
              polls={Array []}
              title="Polls created by other users:"
            />
          </div>
        </Row>
      </Container>,
      "node": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Row>
            <h3>
              Dashboard
            </h3>
            <ul
              className="tabs"
            >
              <li
                className="tab col s3"
              >
                <a
                  href="#my"
                >
                  My Polls
                </a>
              </li>
              <li
                className="tab col s3"
              >
                <a
                  href="#other"
                >
                  Other Polls
                </a>
              </li>
            </ul>
          </Row>,
          <Row>
            <div
              className="col s12"
              id="my"
            >
              <Connect(PollList)
                itemsPerPage={15}
                maxButtons={7}
                polls={Array []}
                title="Polls created by you:"
              />
            </div>
            <div
              className="col s12"
              id="other"
            >
              <Connect(PollList)
                itemsPerPage={15}
                maxButtons={7}
                polls={Array []}
                title="Polls created by other users:"
              />
            </div>
          </Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h3>
                Dashboard
              </h3>,
              <ul
                className="tabs"
              >
                <li
                  className="tab col s3"
                >
                  <a
                    href="#my"
                  >
                    My Polls
                  </a>
                </li>
                <li
                  className="tab col s3"
                >
                  <a
                    href="#other"
                  >
                    Other Polls
                  </a>
                </li>
              </ul>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dashboard",
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="tab col s3"
                  >
                    <a
                      href="#my"
                    >
                      My Polls
                    </a>
                  </li>,
                  <li
                    className="tab col s3"
                  >
                    <a
                      href="#other"
                    >
                      Other Polls
                    </a>
                  </li>,
                ],
                "className": "tabs",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="#my"
                    >
                      My Polls
                    </a>,
                    "className": "tab col s3",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "My Polls",
                      "href": "#my",
                    },
                    "ref": null,
                    "rendered": "My Polls",
                    "type": "a",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="#other"
                    >
                      Other Polls
                    </a>,
                    "className": "tab col s3",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Other Polls",
                      "href": "#other",
                    },
                    "ref": null,
                    "rendered": "Other Polls",
                    "type": "a",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="col s12"
                id="my"
              >
                <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by you:"
                />
              </div>,
              <div
                className="col s12"
                id="other"
              >
                <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by other users:"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by you:"
                />,
                "className": "col s12",
                "id": "my",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "itemsPerPage": 15,
                  "maxButtons": 7,
                  "polls": Array [],
                  "title": "Polls created by you:",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by other users:"
                />,
                "className": "col s12",
                "id": "other",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "itemsPerPage": 15,
                  "maxButtons": 7,
                  "polls": Array [],
                  "title": "Polls created by other users:",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <Row>
            <h3>
              Dashboard
            </h3>
            <ul
              className="tabs"
            >
              <li
                className="tab col s3"
              >
                <a
                  href="#my"
                >
                  My Polls
                </a>
              </li>
              <li
                className="tab col s3"
              >
                <a
                  href="#other"
                >
                  Other Polls
                </a>
              </li>
            </ul>
          </Row>
          <Row>
            <div
              className="col s12"
              id="my"
            >
              <Connect(PollList)
                itemsPerPage={15}
                maxButtons={7}
                polls={Array []}
                title="Polls created by you:"
              />
            </div>
            <div
              className="col s12"
              id="other"
            >
              <Connect(PollList)
                itemsPerPage={15}
                maxButtons={7}
                polls={Array []}
                title="Polls created by other users:"
              />
            </div>
          </Row>
        </Container>,
        "node": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Row>
              <h3>
                Dashboard
              </h3>
              <ul
                className="tabs"
              >
                <li
                  className="tab col s3"
                >
                  <a
                    href="#my"
                  >
                    My Polls
                  </a>
                </li>
                <li
                  className="tab col s3"
                >
                  <a
                    href="#other"
                  >
                    Other Polls
                  </a>
                </li>
              </ul>
            </Row>,
            <Row>
              <div
                className="col s12"
                id="my"
              >
                <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by you:"
                />
              </div>
              <div
                className="col s12"
                id="other"
              >
                <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by other users:"
                />
              </div>
            </Row>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <h3>
                  Dashboard
                </h3>,
                <ul
                  className="tabs"
                >
                  <li
                    className="tab col s3"
                  >
                    <a
                      href="#my"
                    >
                      My Polls
                    </a>
                  </li>
                  <li
                    className="tab col s3"
                  >
                    <a
                      href="#other"
                    >
                      Other Polls
                    </a>
                  </li>
                </ul>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Dashboard",
                },
                "ref": null,
                "rendered": "Dashboard",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li
                      className="tab col s3"
                    >
                      <a
                        href="#my"
                      >
                        My Polls
                      </a>
                    </li>,
                    <li
                      className="tab col s3"
                    >
                      <a
                        href="#other"
                      >
                        Other Polls
                      </a>
                    </li>,
                  ],
                  "className": "tabs",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href="#my"
                      >
                        My Polls
                      </a>,
                      "className": "tab col s3",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "My Polls",
                        "href": "#my",
                      },
                      "ref": null,
                      "rendered": "My Polls",
                      "type": "a",
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href="#other"
                      >
                        Other Polls
                      </a>,
                      "className": "tab col s3",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Other Polls",
                        "href": "#other",
                      },
                      "ref": null,
                      "rendered": "Other Polls",
                      "type": "a",
                    },
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div
                  className="col s12"
                  id="my"
                >
                  <Connect(PollList)
                    itemsPerPage={15}
                    maxButtons={7}
                    polls={Array []}
                    title="Polls created by you:"
                  />
                </div>,
                <div
                  className="col s12"
                  id="other"
                >
                  <Connect(PollList)
                    itemsPerPage={15}
                    maxButtons={7}
                    polls={Array []}
                    title="Polls created by other users:"
                  />
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(PollList)
                    itemsPerPage={15}
                    maxButtons={7}
                    polls={Array []}
                    title="Polls created by you:"
                  />,
                  "className": "col s12",
                  "id": "my",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "itemsPerPage": 15,
                    "maxButtons": 7,
                    "polls": Array [],
                    "title": "Polls created by you:",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(PollList)
                    itemsPerPage={15}
                    maxButtons={7}
                    polls={Array []}
                    title="Polls created by other users:"
                  />,
                  "className": "col s12",
                  "id": "other",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "itemsPerPage": 15,
                    "maxButtons": 7,
                    "polls": Array [],
                    "title": "Polls created by other users:",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
